pkg/contexts/ocm/blobhandler/handlers/generic/mvn/blobhandler.go

```diff
@@ -0,0 +1,166 @@
+package mvn
+
+import (
+	"context"
+	"crypto"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/mandelsoft/vfs/pkg/vfs"
+
+	"github.com/open-component-model/ocm/pkg/contexts/credentials/builtin/mvn/identity"
+	"github.com/open-component-model/ocm/pkg/contexts/ocm/accessmethods/mvn"
+	"github.com/open-component-model/ocm/pkg/contexts/ocm/cpi"
+	"github.com/open-component-model/ocm/pkg/contexts/ocm/cpi/accspeccpi"
+	"github.com/open-component-model/ocm/pkg/contexts/ocm/resourcetypes"
+	"github.com/open-component-model/ocm/pkg/iotools"
+	"github.com/open-component-model/ocm/pkg/logging"
+	"github.com/open-component-model/ocm/pkg/mime"
+	"github.com/open-component-model/ocm/pkg/utils/tarutils"
+)
+
+const BLOB_HANDLER_NAME = "ocm/" + resourcetypes.MVN_ARTIFACT
+
+type artifactHandler struct {
+	spec *Config
+}
+
+func NewArtifactHandler(repospec *Config) cpi.BlobHandler {
+	return &artifactHandler{repospec}
+}
+
+var log = logging.DynamicLogger(identity.REALM)
+
+func (b *artifactHandler) StoreBlob(blob cpi.BlobAccess, resourceType string, hint string, _ cpi.AccessSpec, ctx cpi.StorageContext) (cpi.AccessSpec, error) {
+	// check conditions
+	if b.spec == nil {
+		return nil, nil
+	}
+	mimeType := blob.MimeType()
+	if resourcetypes.MVN_ARTIFACT != resourceType {
+		log.Debug("not a MVN artifact", "resourceType", resourceType)
+		return nil, nil
+	}
+	if mime.MIME_TGZ != mimeType {
+		log.Debug("not a tarball, can't be a complete mvn GAV", "mimeType", mimeType)
+		return nil, nil
+	}
+	if b.spec.Url == "" {
+		return nil, fmt.Errorf("MVN repository url not provided")
+	}
+
+	// setup logger
+	log := log.WithValues("repository", b.spec.Url)
```

Comment:
Should use the logger from the context. `ctx.GetContext().Logger()`

Reply:
`log.` is already a `DynamicLogger` see https://github.com/open-component-model/ocm/blob/maven/access2/pkg/contexts/ocm/blobhandler/handlers/generic/mvn/blobhandler.go#L35